cunumeric/array.py

```diff
@@ -955,12 +955,101 @@ def byteswap(self, inplace=False):
             numpy_array = self.__array__(stacklevel=3).byteswap(inplace=False)
             return self.convert_to_cunumeric_ndarray(numpy_array, stacklevel=3)
 
-    @unimplemented
-    def choose(self, choices, out, mode="raise"):
-        numpy_array = self.__array__(stacklevel=3).choose(
-            choices=choices, out=out, mode=mode
-        )
-        return self.convert_to_cunumeric_ndarray(numpy_array, stacklevel=3)
+    def choose(self, choices, out=None, mode="raise", stacklevel=2):
+        a = self.convert_to_cunumeric_ndarray(self)
+        if out is not None:
+            out = out.convert_to_cunumeric_ndarray(out)
+        # choices can be passed as array or tuple of arrays.
+        # in case of the tuple, we will pass each array separately
+        is_tuple = isinstance(choices, tuple)
+        if is_tuple:
+            n = len(choices)
+            ch_dtype = choices[0].dtype
+        else:
+            n = choices.shape[0]
+            ch_dtype = choices.dtype
+
+        if self.dtype != int:
+            raise TypeError("a array should be ineger type")
+        if mode == "raise":
+            if (a < 0).any() | (a >= n).any():
+                raise ValueError("invalid entry in choice array")
+        elif mode == "wrap":
+            a = a % n
+        elif mode == "clip":
+            a = a.clip(0, n - 1)
+        else:
+            raise ValueError(
+                "mode={mode!r} not understood. Must"
+                " be 'raise', 'wrap', or 'clip'"
+            )
+
+        # we need to broadcast all arrays in choices with
+        # input and output arrays
+        if out is not None:
+            out_shape = broadcast_shapes(a.shape, choices[0].shape, out.shape)
```

Comment:
`choices[0].shape` will not work if choices are passed as an array

Reply:
Is this a restriction of cunumeric?
This seems to work just fine in numpy:
```
a=np.arange(18).reshape((2,3,3))
print(a.shape) #returns (2,3,3)
print(a[0].shape) #returns (3,3)
```

Reply:
You're right, it does work, my mistake.